protocol-definitions/MC.yaml

```diff
@@ -981,3 +981,61 @@ methods:
           since: 2.8
           doc: |
             Member configuration file content (XML or YAML string) of the member or null if the server was not started using a configuration file.
+  - id: 39
+    name: setDiagnosticsConfig
+    since: 2.9
+    doc: |
+      Sets the diagnostics configuration on a running cluster.
+    request:
+      retryable: false
+      partitionIdentifier: -1
+      params:
+        - name: enabled
+          type: boolean
+          nullable: false
+          since: 2.9
+          doc: |
+            Indicates whether diagnostics service is enabled or not
+        - name: outputType
+          type: String
+          nullable: false
+          since: 2.9
+          doc: |
+            The output type for the diagnostics
+        - name: includeEpochTime
+          type: boolean
+          nullable: false
+          since: 2.9
+          doc: |
+            Indicates if the epoch time should be included in the 'top' section
+        - name: maxRolledFileSizeInMB
+          type: int
+          nullable: false
+          since: 2.9
+          doc: |
+            The maximum size in MB for a single file
+        - name: maxRolledFileCount
+          type: int
+          nullable: false
+          since: 2.9
+          doc: |
+            The maximum number of rolling files to keep on disk
+        - name: logDirectory
+          type: String
+          nullable: false
+          since: 2.9
+          doc: |
+            The output directory of the diagnostics log files
+        - name: fileNamePrefix
+          type: String
+          nullable: true
+          since: 2.9
+          doc: |
+            The prefix for the diagnostics file
+        - name: properties
+          type: Map_String_String
+          nullable: true
+          since: 2.9
+          doc: |
+            Properties for the diagnostic plugins
+    response: {}
```

Comment:
Can you comment on the response? E.g. This message changes diagnostics cluster wide and if one member fails, what happens? Which exceptions do we expect?

Reply:
fixed.